File: sol/150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token in '+-*/':
                opnd2 = stack.pop()
                opnd1 = stack.pop()
                if token == '+':
                    stack.append(opnd1 + opnd2)
                elif token == '-':
                    stack.append(opnd1 - opnd2)
                elif token == '*':
                    stack.append(opnd1 * opnd2)
                elif token == '/':
                    stack.append(int(opnd1 / opnd2))
            else:
                stack.append(int(token))
        return stack[-1]

File: sol/150/150.cpp
struct Solution {
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        int op1, op2;
        for (string &s : tokens){
            if (s == "+" || s == "-" || s == "*" || s == "/"){
                op2 = st.top();
                st.pop();
                op1 = st.top();
                st.pop();
                switch (s[0]){
                    case '+':
                        st.push(op1 + op2);
                        break;
                    case '-':
                        st.push(op1 - op2);
                        break;
                    case '*':
                        st.push(op1 * op2);
                        break;
                    case '/':
                        st.push(op1 / op2);
                        break;
                }
            } else {
                st.push(stoi(s));
            }
        }
        return st.top();
    }
};
